plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java

```diff
@@ -27,13 +29,24 @@
 import org.weakref.jmx.guice.MBeanModule;
 
 import java.util.Map;
+import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public class ElasticsearchConnectorFactory
         implements ConnectorFactory
 {
-    ElasticsearchConnectorFactory() {}
+    private final Optional<Module> extension;
```

Comment:
Module is a "collection" already, so you don't need to wrap it in list or optional. You can always use `Modules.EMPTY_MODULE` here. Then logic below is much easier.

Reply:
Applied